sdk/cosmos/src/requests/create_document_builder.rs

```diff
@@ -49,15 +51,19 @@ impl<'a, 'b> CreateDocumentBuilder<'a, 'b> {
         allow_tentative_writes: TenativeWritesAllowance,
         is_upsert: bool => if is_upsert { IsUpsert::Yes } else { IsUpsert::No },
         indexing_directive: IndexingDirective,
-        partition_keys: PartitionKeys => Some(partition_keys),
     }
 }
 
-impl<'a, 'b> CreateDocumentBuilder<'a, 'b> {
-    pub async fn execute<T: Serialize>(
+impl<'a, 'b, 'c> CreateDocumentBuilder<'a, 'b> {
+    async fn execute_internal<DOC, FNPK>(
```

Comment:
Nit: maybe call it `perform_execute`?

Reply:
mmm `perform_execute` seems a bit weird (they are synonyms so it's like saying the same thing twice) but I trust your judgement 👍